cedar-lean/Cedar/Thm/Data/List.lean

```diff
@@ -718,21 +656,117 @@ theorem any_of_mem {f : α → Bool} {x : α} {xs : List α}
   (h₂ : f x) :
   any xs f = true
 := by
-  cases xs <;> simp at h₁
+  cases xs <;> simp only [mem_cons, not_mem_nil] at h₁
   case cons hd tl =>
-    simp [List.any_cons]
+    simp only [any_cons, Bool.or_eq_true, any_eq_true]
     rcases h₁ with h₁ | h₁
-    subst h₁ ; simp [h₂]
-    apply Or.inr ; exists x
+    subst h₁
+    simp only [h₂, true_or]
+    apply Or.inr
+    exists x
+
+/-! ### map and map₁ -/
+
+/--
+  Copied from Mathlib. We can delete this if it gets added to Std.
+-/
+theorem map_congr {f g : α → β} : ∀ {l : List α},
+  (∀ x ∈ l, f x = g x) → map f l = map g l
+  | [], _ => rfl
+  | a :: l, h => by
+    let ⟨h₁, h₂⟩ := forall_mem_cons.1 h
+    rw [map, map, h₁, map_congr h₂]
+
+/--
+  Copied from Mathlib. We can delete this if it gets added to Std.
+-/
+theorem map_pmap_subtype
```

Comment:
Moved here from below. Instead of a dedicated section for "theorems lifted from Mathlib", I put them in the sections where the logically belong (in this case, the "map and map1" section)

Reply:
Conversation moved to https://github.com/cedar-policy/cedar-spec/pull/305#discussion_r1587739841